### DISCUSSION TITLE
Vector point layer as predict feature.

### DISCUSSION THREAD

**Discussion by DemodriverGIS** [opengeos/segment-geospatial#32] [Ideas]
Hello Professor Wu. I'm new at Python GIS. I'm an archeologist and I've been using your sam-geo notebook to segment a concrete type of archeological site in Bolívia from satellite images. The final objective is to get all the areas of the sites (the function mask to vectors you added makes the process easier ty)
The auto mask mode works just fine with almost all my dataset, but some of the sites are not segmented properly or are over-segmented. I have a shape layer with the points of every site, and I'm trying to add the points as prediction elements.  I see that you are creating a branch that adds segmentation with point predictor. It would be amazing if someone finds a way to convert the points of a vector multipoint layer to these predict features.
I'm processing thousands of images that contain all the sites and my solution was to get the boxes of every image and using GEE or Geopandas calculate the relative position of the points within every bbox, then correct the distance considering the pixel resolution of the image (because SAM is not georeferenced by default and asks for a xy coordinate counting pixels). As a result, the points land more or less where they suppose to be, but I'm pretty sure that it isn't the correct or clean way to do it. Maybe it is a too concrete case to implement in your library but I welcome any idea.
Probably my biggest problem is that I don't know how to use SAM with georeferenced elements (as segment-geospatial kind of dose).
Thanks for all your work, I'm learning a lot from your notebooks and tutorials. And thanks in advance to anybody who's going to answer, or spend some time reading this.

_PD: My code is a bit messy at the moment I will attach it to this discussion as soon as it gets showable. v.v_

**Comment by giswqs** [comment]
Good suggestion. I can add vector points as input prompts.
